assets/src/components/v2/departures/section.ts

```diff
@@ -106,37 +111,90 @@ const trimOneBy = (
 // Destructively shift the last above-the-fold departure time in a section to
 // below-the-fold, if possible. Returns `true` if the section was trimmed.
 const trimSection = (section: FoldedSection): boolean => {
-  if (section.type == "folded_section") {
-    const {
-      rows: { aboveFold, belowFold },
-    } = section;
-
-    for (let aboveIndex = aboveFold.length - 1; aboveIndex >= 0; aboveIndex--) {
-      const row = aboveFold[aboveIndex];
-
-      if (row.type == "departure_row") {
-        const trimmedTimeWithCrowding = _.last(row.times_with_crowding);
-
-        if (row.times_with_crowding.length > 1) {
-          // More than one departure time; trim the last one.
-          row.times_with_crowding.splice(row.times_with_crowding.length - 1);
-        } else {
-          // Only one departure time; trim this row entirely.
-          aboveFold.splice(aboveIndex, 1);
-        }
-
-        if (trimmedTimeWithCrowding != null) {
-          belowFold.unshift({
-            ...row,
-            id: _.uniqueId(`${row.id}_`),
-            times_with_crowding: [trimmedTimeWithCrowding],
-          });
-        }
-
-        return true;
+  const {
+    rows: { aboveFold, belowFold },
+  } = section;
+
+  for (let aboveIndex = aboveFold.length - 1; aboveIndex >= 0; aboveIndex--) {
+    const row = aboveFold[aboveIndex];
+
+    if (isDepartureRow(row)) {
+      const timesWithCrowding = row.times_with_crowding;
+      const trimmedTimeWithCrowding = _.last(timesWithCrowding);
+
+      if (timesWithCrowding.length > 1) {
+        // More than one departure time; trim the last one.
+        timesWithCrowding.splice(timesWithCrowding.length - 1);
+      } else {
+        // Only one departure time; trim this row entirely.
+        aboveFold.splice(aboveIndex, 1);
       }
+
+      if (trimmedTimeWithCrowding != null) {
+        belowFold.unshift({
+          ...row,
+          id: _.uniqueId(`${row.id}_`),
+          times_with_crowding: [trimmedTimeWithCrowding],
+        });
+      }
+
+      return true;
     }
   }
 
   return false;
 };
+
+const trimDestinationGroupingSection = (section: FoldedSection): boolean => {
+  const {
+    rows: { aboveFold, belowFold },
+    direction_trim_priority: directionTrimPriority,
+  } = section;
+
+  const isBidirectional =
+    aboveFold.some((row) => isDepartureRow(row) && row.direction_id == 0) &&
+    aboveFold.some((row) => isDepartureRow(row) && row.direction_id == 1);
+
+  // If there's only one direction, treat it as a timed list and trim the furthest time
+  const trimmedRow = isBidirectional
+    ? trimBidirectionDestinationGrouping(aboveFold, directionTrimPriority)
+    : _.last(aboveFold);
+
+  if (isDepartureRow(trimmedRow)) {
+    aboveFold.splice(aboveFold.indexOf(trimmedRow), 1);
+    belowFold.unshift({
```

Comment:
Since sections with destination-based grouping never use "later departures", I think there is some simplification we could do here. We don't technically need to shift trimmed departures below the fold, we just need to discard them.